examples/data-objects/inventory-app/src/view/inventoryList.tsx

```diff
@@ -27,7 +27,9 @@ const schemaPolicy = {
 };
 
 export const MainView: React.FC<{ tree: ISharedTree }> = ({ tree }) => {
-	const inventory: Inventory = useTree(tree, schemaPolicy);
+	const root: RootField = useTree(tree, schemaPolicy);
+	// TODO: value fields like `root` which always contain exactly one value should have a nicer API for accessing that child, like `.child`.
+	const inventory: Inventory = root[0];
```

Comment:
Having correct strong types informs the user what they get: if they are wrong their code won't build.

If we unwrap here then it's impossible for users of this API to change the root of the document. Imagine having an optional root field (so you can load an empty document) and just getting back "undefined" with no possible way to insert content into the document: not very useful.

At this point in API development, I don't think we want to hide important concepts (like fields) implicitly, especially in places where there is no practical way to get to them if/when you need to if they are hidden. I don't want to try and make the API more intuitive and clean looking initially just to prevent a user from having to learn important concepts they will need to learn anyway while also making users which need to do specific actions (like editing the root) have a real hard time doing so (not use the schematized tree, and call tree.context.root yourself, then manually edit that: good luck discovering how to do that workaround).

I think erroring on the side of explicitness, discoverability and completeness is better than erroring on the side of conciseness and intuitiveness for new users, especially since we have strong typing so developers are quickly shown their mistakes when they guess wrong. Examples and APIs which provide an illusion of elegance are a poor tool for actually educating users, and a good way to miss things and fail to learn key concepts leading to getting stuck later.